opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A float with the electric charge of the selected particles in Coulomb
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        w, q = self.get_particle( var_list=['w', 'charge'], species=species,
+                                 t=t, iteration=iteration )
+        # Calculate charge
+        charge = np.sum(w) * q
+        # Return the result
+        return( charge )
+
+    def get_divergence( self, t=None, iteration=None, species=None,
+                        select=None ):
+        """
+        Calculate the divergence of the selected particles.
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A tuple with:
+        - divergence in x plane in rad
+        - divergence in y plane in rad
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        ux, uy, uz, w = self.get_particle( var_list=['ux', 'uy', 'uz', 'w'],
+                                           t=t, iteration=iteration,
+                                           species=species )
+        # Calculate diveregence
+        div_x = wstd( ux / uz, w )
+        div_y = wstd( uy / uz, w )
+        # Return the result
```

Comment:
What about doing
```python
div_x = wstd( np.arctan2(ux, uz), w )
div_y = wstd( np.arctan2(ux, uz), w )
```
so that it is also valid for large deflection angles, and does not crash for uz=0.